### ISSUE TITLE
Compilation errors for `make cross-riscv`

### ISSUE THREAD

**Issue by jan-heck** [physical-computation/sunflower-simulator#108]
**Describe the bug**
I am getting compilation errors when compiling `make cross-riscv` that prevent successful compilation.

What I did to narrow down the scope of the issues:

- make cross-superh works
- make works 
- git cloned a fresh repository, followed GB3 course notes pg 25f to setup; same results

The problem then is that `make cross-riscv` fails as follows:

### Compilation Error 1)
After ~5min of compilation, compilation fails with `cc1` not found. Apparently, cc1 is the gcc-internal preprocessor step executable.

I fixed this by adding
`export PATH="$HOME/git/sunflower-simulator/sunflower-toolchain/tools/riscv/libexec/gcc/riscv32-elf/8.2.0:$PATH" # cc1`
 to my path, allowing compilation to continue beyond this point

### Compilation Error 2)
With the above fix, compilation fails at a later point, with an assembler error and  missing #includes:
```
[snip]
riscv32-elf-cc -B/home/arch/git/sunflower-simulator/sunflower-toolchain/tools/source/newlib-2.5.0.20170922/objdir/riscv32-elf/rv32i/ilp32/newlib/ -isystem /home/arch/git/sunflower-simulator/sunflower-toolchain/tools/source/newlib-2.5.0.20170922/objdir/riscv32-elf/rv32i/ilp32/newlib/targ-include -isystem /home/arch/git/sunflower-simulator/sunflower-toolchain/tools/source/newlib-2.5.0.20170922/newlib/libc/include -B/home/arch/git/sunflower-simulator/sunflower-toolchain/tools/source/newlib-2.5.0.20170922/objdir/riscv32-elf/rv32i/ilp32/libgloss/riscv32 -L/home/arch/git/sunflower-simulator/sunflower-toolchain/tools/source/newlib-2.5.0.20170922/objdir/riscv32-elf/rv32i/ilp32/libgloss/libnosys -L/home/arch/git/sunflower-simulator/sunflower-toolchain/tools/source/newlib-2.5.0.20170922/libgloss/riscv32  -march=rv32i -mabi=ilp32 -DPACKAGE_NAME=\"newlib\" -DPACKAGE_TARNAME=\"newlib\" -DPACKAGE_VERSION=\"2.5.0\" -DPACKAGE_STRING=\"newlib\ 2.5.0\" -DPACKAGE_BUGREPORT=\"\" -DPACKAGE_URL=\"\" -I. -I../../../../../../../newlib/libc/argz -fno-builtin      -g -O2  -march=rv32i -mabi=ilp32 -c -o lib_a-argz_append.o `test -f 'argz_append.c' || echo '../../../../../../../newlib/libc/argz/'`argz_append.c
riscv32-elf-cc -B/home/arch/git/sunflower-simulator/sunflower-toolchain/tools/source/newlib-2.5.0.20170922/objdir/riscv32-elf/rv32i/ilp32/newlib/ -isystem /home/arch/git/sunflower-simulator/sunflower-toolchain/tools/source/newlib-2.5.0.20170922/objdir/riscv32-elf/rv32i/ilp32/newlib/targ-include -isystem /home/arch/git/sunflower-simulator/sunflower-toolchain/tools/source/newlib-2.5.0.20170922/newlib/libc/include -B/home/arch/git/sunflower-simulator/sunflower-toolchain/tools/source/newlib-2.5.0.20170922/objdir/riscv32-elf/rv32i/ilp32/libgloss/riscv32 -L/home/arch/git/sunflower-simulator/sunflower-toolchain/tools/source/newlib-2.5.0.20170922/objdir/riscv32-elf/rv32i/ilp32/libgloss/libnosys -L/home/arch/git/sunflower-simulator/sunflower-toolchain/tools/source/newlib-2.5.0.20170922/libgloss/riscv32  -march=rv32i -mabi=ilp32 -DPACKAGE_NAME=\"newlib\" -DPACKAGE_TARNAME=\"newlib\" -DPACKAGE_VERSION=\"2.5.0\" -DPACKAGE_STRING=\"newlib\ 2.5.0\" -DPACKAGE_BUGREPORT=\"\" -DPACKAGE_URL=\"\" -I. -I../../../../../../../newlib/libc/argz -fno-builtin      -g -O2  -march=rv32i -mabi=ilp32 -c -o lib_a-argz_count.o `test -f 'argz_count.c' || echo '../../../../../../../newlib/libc/argz/'`argz_count.c
Assembler messages:
Fatal error: invalid -march= option: `rv32i'
make[10]: *** [Makefile:393: lib_a-dummy.o] Error 1
make[10]: *** Waiting for unfinished jobs....
In file included from /home/arch/git/sunflower-simulator/sunflower-toolchain/tools/source/newlib-2.5.0.20170922/newlib/libc/include/sys/errno.h:11,
                 from /home/arch/git/sunflower-simulator/sunflower-toolchain/tools/source/newlib-2.5.0.20170922/newlib/libc/include/errno.h:9,
                 from ../../../../../../../newlib/libc/argz/argz_add_sep.c:7:
/home/arch/git/sunflower-simulator/sunflower-toolchain/tools/source/newlib-2.5.0.20170922/newlib/libc/include/sys/reent.h:14:10: fatal error: stddef.h: No such file or directory
 #include <stddef.h>
          ^~~~~~~~~~
compilation terminated.
make[10]: *** [Makefile:405: lib_a-argz_add_sep.o] Error 1
In file included from /home/arch/git/sunflower-simulator/sunflower-toolchain/tools/source/newlib-2.5.0.20170922/newlib/libc/include/sys/errno.h:11,
                 from /home/arch/git/sunflower-simulator/sunflower-toolchain/tools/source/newlib-2.5.0.20170922/newlib/libc/include/errno.h:9,
                 from /home/arch/git/sunflower-simulator/sunflower-toolchain/tools/source/newlib-2.5.0.20170922/newlib/libc/include/argz.h:10,
                 from ../../../../../../../newlib/libc/argz/argz_count.c:8:
/home/arch/git/sunflower-simulator/sunflower-toolchain/tools/source/newlib-2.5.0.20170922/newlib/libc/include/sys/reent.h:14:10: fatal error: stddef.h: No such file or directory
 #include <stddef.h>
          ^~~~~~~~~~
compilation terminated.
In file included from /home/arch/git/sunflower-simulator/sunflower-toolchain/tools/source/newlib-2.5.0.20170922/newlib/libc/include/sys/errno.h:11,
                 from /home/arch/git/sunflower-simulator/sunflower-toolchain/tools/source/newlib-2.5.0.20170922/newlib/libc/include/errno.h:9,
                 from /home/arch/git/sunflower-simulator/sunflower-toolchain/tools/source/newlib-2.5.0.20170922/newlib/libc/include/argz.h:10,
                 from ../../../../../../../newlib/libc/argz/argz_append.c:7:
/home/arch/git/sunflower-simulator/sunflower-toolchain/tools/source/newlib-2.5.0.20170922/newlib/libc/include/sys/reent.h:14:10: fatal error: stddef.h: No such file or directory
 #include <stddef.h>
          ^~~~~~~~~~
compilation terminated.
make[10]: *** [Makefile:417: lib_a-argz_count.o] Error 1
make[10]: *** [Makefile:411: lib_a-argz_append.o] Error 1
In file included from /home/arch/git/sunflower-simulator/sunflower-toolchain/tools/source/newlib-2.5.0.20170922/newlib/libc/include/sys/errno.h:11,
                 from /home/arch/git/sunflower-simulator/sunflower-toolchain/tools/source/newlib-2.5.0.20170922/newlib/libc/include/errno.h:9,
                 from /home/arch/git/sunflower-simulator/sunflower-toolchain/tools/source/newlib-2.5.0.20170922/newlib/libc/include/argz.h:10,
                 from ../../../../../../../newlib/libc/argz/argz_add.c:7:
/home/arch/git/sunflower-simulator/sunflower-toolchain/tools/source/newlib-2.5.0.20170922/newlib/libc/include/sys/reent.h:14:10: fatal error: stddef.h: No such file or directory
 #include <stddef.h>
          ^~~~~~~~~~
compilation terminated.
```

I assume this is the root cause: `Assembler messages:
Fatal error: invalid -march= option: 'rv32i'`

**To Reproduce**
Change setup.conf to risc-v settings (attached)
`make cross-riscv`

**Expected behavior**
Successful build.

**Host OS (please complete the following information):**
Arch Linux x64
```
~ uname -a
Linux nbpb 5.2.9-1-ck-skylake #1 SMP PREEMPT Fri Aug 16 17:34:48 EDT 2019 x86_64 GNU/Linux
```

**You local changes (please complete the following information):**
Only setup.conf (attached in full), as instructed by GB3 course notes / sunflower readme:
```
SUNFLOWERROOT	= /home/arch/git/sunflower-simulator

OSTYPE		= linux
MACHTYPE	= i386
#TARGET		= arm
#TARGET		= superH
TARGET		= riscv
#TARGET-ARCH-FLAGS = -DM32 # What does this belong to?
#TARGET-ARCH	= arm-none-eabi
#TARGET-ARCH	= sh-elf
TARGET-ARCH	= riscv32-elf
GMAKE		= /usr/bin/make


#
#       Specifies the architecture used. Uncomment for riscv builds.
#
ADDITIONAL_ARCH_FLAGS = --with-cpu=rv32ifd
```

```
git diff in ~/git/sunflower-simulator:
diff --git a/sunflower-toolchain b/sunflower-toolchain
index 1e473d9..b74c5dc 160000
--- a/sunflower-toolchain
+++ b/sunflower-toolchain
@@ -1 +1 @@
-Subproject commit 1e473d9a5aec1300041ccf2b02d50de33c546541
+Subproject commit b74c5dc3457adf86fe428605fac176f639022b1d-dirty

git diff in ~/git/sunflower-simulator/sunflower-toolchain
diff --git a/conf/setup.conf b/conf/setup.conf
index c47b6b9..a2dff4c 100755
--- a/conf/setup.conf
+++ b/conf/setup.conf
@@ -1,19 +1,24 @@
 ##
 ## You will want to change the following to suit your setup:
 ##
-SUNFLOWERROOT  = /tmp/sunflower-toolchain-github-clone
+SUNFLOWERROOT  = /home/arch/git/sunflower-simulator
 
 OSTYPE   = linux
 MACHTYPE = i386
-TARGET   = arm
+#TARGET    = arm
 #TARGET    = superH
-#TARGET    = riscv
-TARGET-ARCH-FLAGS = -DM32
-TARGET-ARCH  = arm-none-eabi
+TARGET   = riscv
+#TARGET-ARCH-FLAGS = -DM32 # What does this belong to?
+#TARGET-ARCH = arm-none-eabi
 #TARGET-ARCH = sh-elf
-#TARGET-ARCH = riscv32-elf
+TARGET-ARCH  = riscv32-elf
 GMAKE    = /usr/bin/make
 
+#
+#       Specifies the architecture used. Uncomment for riscv builds.
+#
+ADDITIONAL_ARCH_FLAGS = --with-cpu=rv32ifd
+
 #
 #  On newer versions of macOS (High Sierra), you will need to install gcc
 #  (as opposed to using the gcc alias on modern macOS that is


~/git/sunflower-simulator git remote -v
origin	https://github.com/phillipstanleymarbell/sunflower-simulator.git (fetch)
origin	https://github.com/phillipstanleymarbell/sunflower-simulator.git (push)

```

*** Other notes ***
The line in setup.conf:
`#TARGET-ARCH-FLAGS = -DM32`
appears a bit orphaned to me, it isn't clear what architecture(s) it relates to. It isn't mentioned in any of the documentation afaict. Grepping the source makes me believe this is only relevant to arm-based architectures? I assume this is unrelated, in which case I can open a different issue (or PR once somebody enlightens me what it does).

In any case, tried compiling with and without commenting it out, no difference.

*** Attachments ***
[setup.conf.txt](https://github.com/physical-computation/sunflower-simulator/files/3560002/setup.conf.txt)

**Comment by jan-heck** [comment]
I just realised my remotes are still github.com/phillipstanleymarbell, I assumed these would be automatically transferred?

I will change to physcomplab

**Comment by phillipstanleymarbell** [comment]
Did you add  `$SUNFLOWERROOT/tools/bin` to your path (Step 3 of https://github.com/phillipstanleymarbell/Sunflower-toolchain)?

**Comment by jan-heck** [comment]
Yes

**Comment by jan-heck** [comment]
The version from `https://github.com/physical-computation/sunflower-simulator.git` does compile correctly with the same settings. I apologise for the noise. I thought the remote would be shifted to the new one automatically, should have checked beforehand that this was true.

I was able to compile and run/emulate the riscv bubblesort from https://github.com/physical-computation-measurements/issue-88 

Closing.
